test/functional/test-cookies.js

```diff
@@ -109,18 +109,18 @@ describe('cookies', () => {
     test('cdn.ampproject.org', 'ampproject.org', true, true);
     expect(() => {
       test('cdn.ampproject.org', 'ampproject.org', true);
-    }).to.throw(/Should never attempt to set cookie on proxy origin\: c\&1/);
+    }).to.throw(/Should never attempt to set cookie on proxy origin.*\: c\&1/);
```

Comment:
...could have just changed to match line 126 instead, since the full message is:
Should never attempt to set cookie on proxy origin. (in depth check): c&1

Would appreciate close review by someone who knows why lines 112, 115, 118, 123 were originally written differently from 126, 129, 132.

Reply:
The reason this is happening is because of the protocol used. We need to update the regex to match the `:` part of the origin, too.

Reply:
(Assuming you meant "port" rather than "protocol".)
Ok, fixed that, to allow the specification of any port, or none.

Reply:
I also put the unit tests back the way they were.

Reply:
Why would we ever need to check against a nonstandard port? Surely we'd never expose one on the AMP Cache?

Reply:
One would hope not.
Though port 8000 is hard-coded on line 88 of this file. I tested removing ":8000" from that line, and changing the regex in src/config.js back to /^https:\/\/([a-zA-Z0-9_-]+\.)?cdn\.ampproject\.org$/ and all tests in this file pass. I've not pushed that yet, because it makes me curious to know why port 8000 was used in the test in the first place?
@jridgewell is it safe to remove ":8000" from line 88 but then add a test that should fail if given an unusual port?